### PR TITLE
adds more values for userAgent string

### PR DESCRIPTION
**Pull request by ladanazita**
This expands on the concatenated `userAgent` string. Pulls in values from the `context`and as best as I could, I attempted to create a similar value for a `userAgent`. 

The idea is to get Mobile Device info, Mobile Device Branding, and Operating System Version values to populate within the *Audiences > Devices and Network > Overview* :

Currently outputs: `Segment/2.0 (iPhone OS ; CPU like iPhone OS 9.3; en-US) AppleWebKit/600.1.4 (KHTML, like Gecko) Apple; Version 9.3`.

You can see in the gif above that Operating System and Operating System Version aren't populating. I am hesitant to hard code any further information that isn't consistent across the board, and it [looks like](http://www.webapps-online.com/online-tools/user-agent-strings/dv/operatingsystem51849/ios) the (KHTML, like Gecko) value is consistent across the board.

The rest I pull from what we automatically collected in our iOS SDK:

```
form.ua = 'Segment/2.0 (' + osName + ' ; CPU like ' + osName + ' ' + osVersion + '; '+ locale + ') AppleWebKit/600.1.4 (KHTML, like Gecko) ' + deviceManufacturer + '; ' + 'Version ' + osVersion;
```

This is a temporary solution until the iOS SDK has the ability to generate a more accurate `userAgent` value. 

@f2prateek @hankim813

### CODE REVIEW

**Comment by hankim813** [comment]
@ladanazita have you done end 2 end testing for this? Since these tests can hit GA's actual endpoints, could you verify in our end tool that this change works?

**Comment by f2prateek** [comment]
I'm confused - does this actually populate in GA correctly or not?

**Comment by ladanazita** [comment]
@hankim813 @f2prateek I will run the tests again and provide a screen shot. Han brought up a good point that I should add this for the classic GA as well so I will resubmit. 

Thanks!

**Comment by ladanazita** [comment]
Ok new PR fixes the `OS version` and here is a GIF to show how it appears in GA - the `(not set)` was from the first test and the `9_3` after some corrections. 

To test this populating in GA, I changed the tracking id to a new GA instance so as not to confuse any other test calls coming in.

screen shot 2016-10-04 at 5 46 34 pm

File: test/fixtures/page-app.json [hunk 1]
```diff
@@ -27,14 +31,14 @@
   "output": {
     "cid": 2710159508,
     "dh": "segment.io",
-    "tid": "UA-27033709-11",
+    "tid": "UA-27033709-23",
```
Comment (f2prateek) [inline test/fixtures/page-app.json]: We should change this back now that testing is done?

**Comment by f2prateek** [comment]
LGTM otherwise

**Comment by ladanazita** [comment]
@f2prateek I changed the fixture for that `page` call to default to the *mobile tracking id* - so the test would fail if the output took the server-side tracking id.

 This fix is meant to account for when the `userAgent` string is *not present* && the *device context is present*, create a `userAgent` string using the device context automatically collected by the mobile library.

**Comment by f2prateek** [comment]
In confused aa to why the value for the tid matters? What was wrong the old one?

**Comment by ladanazita** [comment]
Nothing _wrong_ with it, I just feel like that is a more accurate fixture to test against for this fix. I can change it back and remove:

```
  "library": {
         "name": "analytics-ios",
         "version": "3.5.0"
     },
```

**Comment by sperand-io** [comment]
Aside: FWIW, classic Google Analytics has been deprecated and all [GA properties have been migrated to be able accept data via the universal measurement protocol](https://developers.google.com/analytics/devguides/collection/upgrade/).

We should actually remove support for using the classic implementation at some point — there's a lot of code we can delete there! And it's theoretically completely backwards compat, we'd just have to do a bit of end to end testing to ensure. (we have some ex-classic properties in our account we can verify with)

**Comment by f2prateek** [comment]
@ladanazita no worries, I still don't really understand but let's keep it as is since you've tested it end to end!

**Comment by ladanazita** [comment]
Is this good to 🔀  and 🚢 ?  @f2prateek @sperand-io @hankim813

**Comment by hankim813** [comment]
@ladanazita Yep! Thanks for the PR :) I'll deploy this guy first thing in the morning!

**Pull request merged**